src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js

```diff
@@ -66,6 +67,19 @@ class BaseDrawerNavigator extends Component {
                     ),
                     swipeEdgeWidth: 500,
                 }}
+                screenListeners={{
+                    state: (e) => {
+                        const state = e.data.state;
+
+                        // Get the history that has drawer type to get the status.
+                        const hasDrawerHistory = _.find(state.history || [], h => h.type === 'drawer');
+
+                        // hasDrawerHistory will has undefined value if the route drawer is equal to initial route drawer.
+                        // Using the default status if hasDrawerHistory is undefined and get the status from the current route
+                        // if the value provided.
```

Comment:
Also don't understand this comment... where does the default status coming from ?

Reply:
The status comes from the route state.

Reply:
Sorry, I'm not sure if I'm following that...

Going to try to explain in my own words and you let me know if I'm on the right track...

The item that we are getting from the `history` is a `route`. That `route` has a `status` that refers to the state of the drawer.

The `e.data.state` has a `default` property to indicate the drawer status as well.

Are these drawer states documented somewhere?

Reply:
It's not well documented on the docs https://reactnavigation.org/docs/navigation-events/#state.

Here are the example from `data.state`
```
{
    "stale": false,
    "type": "drawer",
    "key": "drawer-o_B0nHlaCG0jGCWESf5PA",
    "index": 0,
    "routeNames": [
        "Report"
    ],
    "history": [
        {
            "type": "route",
            "name": "Report"
        },
        {
            "type": "route",
            "name": "Report"
        },
        {
            "type": "route",
            "name": "Report"
        }
    ],
    "routes": [
        {
            "name": "Report",
            "params": {
                "reportID": "90420013"
            },
            "key": "Report-oiQ55dmXeZgHMcLsx0rmh"
        }
    ],
    "default": "closed"
}
```
```
{
    "stale": false,
    "type": "drawer",
    "key": "drawer-o_B0nHlaCG0jGCWESf5PA",
    "index": 0,
    "routeNames": [
        "Report"
    ],
    "history": [
        {
            "type": "route",
            "name": "Report"
        },
        {
            "type": "route",
            "name": "Report"
        },
        {
            "type": "drawer",
            "status": "open"
        }
    ],
    "routes": [
        {
            "name": "Report",
            "params": {
                "reportID": "92055915"
            },
            "key": "Report-YVI0AsQPNmKDO-oKjc-Sl"
        }
    ],
    "default": "closed"
}
```

The `default` value is what we set from `defaultStatus` props of `Drawer.Navigator`.